cpp/cmake/thirdparty/get_faiss.cmake

```diff
@@ -24,22 +24,34 @@ function(find_and_configure_faiss)
         LIBRARY_NAMES faiss
     )
 
+    set(BUILD_SHARED_LIBS OFF)
+    if (NOT BUILD_STATIC_LIBS)
+        set(BUILD_SHARED_LIBS ON)
+    endif()
+
     rapids_cpm_find(FAISS ${PKG_VERSION}
-        GLOBAL_TARGETS  faiss
+        GLOBAL_TARGETS   faiss FAISS::FAISS
+        BUILD_EXPORT_SET raft-exports
         CPM_ARGS
           GIT_REPOSITORY  https://github.com/facebookresearch/faiss.git
           GIT_TAG         ${PKG_PINNED_TAG}
           OPTIONS
             "FAISS_ENABLE_PYTHON OFF"
-            "BUILD_SHARED_LIBS OFF"
+            "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
             "CUDAToolkit_ROOT ${CUDAToolkit_LIBRARY_DIR}"
             "FAISS_ENABLE_GPU ON"
             "BUILD_TESTING OFF"
             "CMAKE_MESSAGE_LOG_LEVEL VERBOSE"
     )
 
     if(FAISS_ADDED)
```

Comment:
Since `faiss` is required when including certain RAFT headers, we should make sure it is in the `target_link_libraries` of raft as PUBLIC